code-review-utils.el

```diff
@@ -236,23 +236,36 @@ using COMMENTS."
 
 (defun code-review-utils-pr-from-url (url)
   "Extract a pr alist from a pull request URL."
-  (cond
-   ((string-prefix-p "https://gitlab.com" url)
-    (save-match-data
-      (and (string-match ".*/\\(.*\\)/\\(.*\\)/-/merge_requests/\\([0-9]+\\)" url)
-           (a-alist 'num (match-string 3 url)
-                    'repo (match-string 2 url)
-                    'owner (match-string 1 url)
-                    'forge 'gitlab
-                    'url url))))
-   ((string-prefix-p "https://github.com" url)
-    (save-match-data
-      (and (string-match ".*/\\(.*\\)/\\(.*\\)/pull/\\([0-9]+\\)" url)
-           (a-alist 'num   (match-string 3 url)
-                    'repo  (match-string 2 url)
-                    'owner (match-string 1 url)
-                    'forge 'github
-                    'url url))))))
+  (let ((gitlab-base-url "gitlab.com")
+        (github-base-url "github.com"))
+    (cond
+     ((string-prefix-p (format "https://%s" gitlab-base-url) url)
+      (save-match-data
+        (let ((res-with-subgroup
+               (and (string-match (format "https://%s/\\(.*\\)/\\(.*\\)/\\(.*\\)/-/merge_requests/\\([0-9]+\\)" gitlab-base-url) url)
```

Comment:
Can we allow to custom numbers of subgroups? Something like
```elisp
(let ((url "https://gitlab.com/owner/group/subgroup1/subgroup2/subgroup3/project/-/merge_requests/1")
      (pattern "https://gitlab.com/\\([^/]*\\)/\\(.*\\)/-/merge_requests/\\([0-9]+\\)"))
  (when (string-match pattern url)
    (list 'num (match-string 3 url)
          'repo (string-replace "/" "%2F" (match-string 2 url))
          'owner (match-string 1 url))))
```

Reply:
Nice! for sure